x-pack/plugins/security_solution/public/management/components/endpoint_responder/get_running_processes_action.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useEffect, useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiBasicTable } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ActionDetails, RunningProcessesEntry } from '../../../../common/endpoint/types';
+import { useGetActionDetails } from '../../hooks/endpoint/use_get_action_details';
+import { EndpointCommandDefinitionMeta } from './types';
+import { CommandExecutionComponentProps } from '../console/types';
+import { useSendGetEndpointRunningProcessesRequest } from '../../hooks/endpoint/use_send_get_endpoint_running_processes_request';
+
+// @ts-expect-error TS2769
+const StyledEuiBasicTable = styled(EuiBasicTable)`
+  .euiTable {
+    background-color: ${({ theme: { eui } }) => eui.euiPageBackgroundColor};
+  }
+  .euiTableHeaderCell {
+    border-bottom: ${(props) => props.theme.eui.euiBorderThin};
+    .euiTableCellContent__text {
+      font-weight: 400;
+    }
+  }
+  .euiTableRowCell {
+    border-top: none !important;
+    border-bottom: none !important;
+  }
+`;
+
+export const GetRunningProcessesActionResult = memo<
+  CommandExecutionComponentProps<
+    { comment?: string },
+    {
+      actionId?: string;
+      actionRequestSent?: boolean;
+      completedActionDetails?: ActionDetails<RunningProcessesEntry>;
+    },
+    EndpointCommandDefinitionMeta
+  >
+>(({ command, setStore, store, status, setStatus, ResultComponent }) => {
+  const endpointId = command.commandDefinition?.meta?.endpointId;
+  const { actionId, completedActionDetails } = store;
+
+  const isPending = status === 'pending';
+  const actionRequestSent = Boolean(store.actionRequestSent);
+
+  const getRunningProcessesApi = useSendGetEndpointRunningProcessesRequest();
+
+  const { data: actionDetails } = useGetActionDetails<RunningProcessesEntry>(actionId ?? '-', {
+    enabled: Boolean(actionId) && isPending,
+    refetchInterval: isPending ? 3000 : false,
+  });
+
+  // Send get running processes request if not yet done
+  useEffect(() => {
+    if (!actionRequestSent && endpointId) {
+      getRunningProcessesApi.mutate({
+        endpoint_ids: [endpointId],
+        comment: command.args.args?.comment?.[0],
+      });
+
+      setStore((prevState) => {
+        return { ...prevState, actionRequestSent: true };
+      });
+    }
+  }, [actionRequestSent, command.args.args?.comment, endpointId, getRunningProcessesApi, setStore]);
+
+  // If get running processes request was created, store the action id if necessary
+  useEffect(() => {
+    if (getRunningProcessesApi.isSuccess && actionId !== getRunningProcessesApi.data.data.id) {
```

Comment:
need to add an `else` to this `if()` statement for cases where the action request failed. In that case, the command will need to be set to "complete" and an error message shown (whatever the message is on the `getRunningProcessesApi` call.

_(FYI: this problem exists today in Isolate and release - I have opened an issue to track and fix those at some point. see OLM issue 4260 )_